File: 73.矩阵置零.py
#
# @lc app=leetcode.cn id=73 lang=python3
#
# [73] 矩阵置零
#

# @lc code=start
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        rows, cols = [], []
        m = len(matrix)
        n = len(matrix[0])
        for i in range(m):
            for j in range(n):
                if matrix[i][j] == 0:
                    rows.append(i)
                    cols.append(j)
        for row in rows:
            for i in range(n):
                matrix[row][i] = 0
        for col in cols:
            for j in range(m):
                matrix[j][col] = 0
# @lc code=end

File: 73.矩阵置零.cpp
/*
 * @lc app=leetcode.cn id=73 lang=cpp
 *
 * [73] 矩阵置零
 */

// @lc code=start
class Solution
{
public:
    void setZeroes(vector<vector<int>> &matrix)
    {
        vector<int> rows;
        vector<int> cols;
        int m = matrix.size();
        int n = matrix[0].size();
        for (int i = 0; i < m; i++)
            for (int j = 0; j < n; j++)
                if (matrix[i][j] == 0)
                {
                    rows.push_back(i);
                    cols.push_back(j);
                }
        for (int row : rows)
            for (int i = 0; i < n; i++)
                matrix[row][i] = 0;
        for (int col : cols)
            for (int i = 0; i < m; i++)
                matrix[i][col] = 0;
    }
};
// @lc code=end
